### DISCUSSION TITLE
How to exclude folders?

### DISCUSSION THREAD

**Discussion by angelus2014** [PatrickF1/fzf.fish#257] [Q&A] [answered]
Hello,

First off, please excuse if this has been addressed before or that I'm just missing something, I've just started with FZF. I'm on a Mac and when using fzf.fish I want to exclude the "Library" folder. I've tried with creating a `.gitignore` file with `Library/` inside it. As I understand, fzf uses fd instead of find and as such should use this `.gitignore` file. It doesn't. I also tried with `set fzf_fd_opts --hidden --exclude=Library/`, again no joy.
I looked inside the fzf_key_bindings.fish file and there I read:
```
test -n "$FZF_CTRL_T_COMMAND"; or set -l FZF_CTRL_T_COMMAND "
command find -L \$dir -mindepth 1 \\( -path \$dir'*/\\.*' -o -fstype 'sysfs' -o -fstype 'devfs' -o -fstype 'devtmpfs' \\) -prune \
```
Does this mean that it isn't using fd, but find??

---
With regards,

Angelo Machils

**Comment by PatrickF1** [comment]
Hi Angelo, are you referencing this file? https://github.com/junegunn/fzf/blob/master/shell/key-bindings.fish
That's a different thing. That's the shell integration that ships with fzf. You can read about the differences here https://github.com/PatrickF1/fzf.fish/wiki/Prior-Art.

**Reply by angelus2014** [reply]
Thank you for the reply and it's clear. But I don't understand why the directory exclusion is not working. It's not working with creating a `.gitignore` or `.ignore` file and it's not working with setting the `set fzf_fd_opts --hidden --exclude=Library/`. These answers I have found while looking into my question before asking it here.

**Reply by PatrickF1** [reply]
Hi, updating my gitignore and passing in `--eclude=Library/` works perfectly fine for me. Are you sure you are executing fzf.fish's search file function and not the one from https://github.com/junegunn/fzf/blob/master/shell/key-bindings.fish?

**Comment by angelus2014** [comment]
Hello,

On 11 Aug 2022, at 22:05, Patrick wrote:

> Hi, updating my gitignore and passing in `--eclude=Library/` works perfectly fine for me. Are you sure you are executing fzf.fish's search file function and not the one from https://github.com/junegunn/fzf/blob/master/shell/key-bindings.fish?

Well, I’m starting it with `CTRL-ALT-T`, which should be the correct way to invoke it.

**Comment by PatrickF1** [comment, marked as answer]
CTRL+ALT+T is definitely the wrong keybinding. fzf.fish uses CTRL+ALT+F
https://github.com/patrickf1/fzf.fish#file-paths.

**Comment by angelus2014** [comment]
Hello,

On 14 Aug 2022, at 2:17, Patrick wrote:

> CTRL+ALT+T is definitely the wrong keybinding. fzf.fish uses CTRL+ALT+F
> https://github.com/patrickf1/fzf.fish#file-paths.

With this key combination, it’s still showing results from the Library folder…

[edit] Oeps, the `opts` setting wasn't active. I should put it somewhere permanent. Apparently, the .gitignore setting doesn't work or it needs both settings to actually work.

Thank you!

---
With regards,

Angelo Machils

**Reply by PatrickF1** [reply]
> Apparently, the .gitignore setting doesn't work or it needs both settings to actually work.

That's an fd question. You'll have to ask them or just read the fd docs more in depth. But in general, since your library folder is in your home directory which is not a git repository, it's tricky.
